### PR TITLE
spec: Update and simplify packaging for Fedora and add EL8 support

### PR DESCRIPTION
**Pull request by Conan-Kudo**
The packaging was a tad bit more complicated than it needed to be for supporting Fedora and openSUSE equally.

This change simplifies it by bringing it up to date with latest packaging practices in Fedora and adding the macro to enable the Python dependency generator for EL8 so that dependencies are correctly auto-generated when building a package for EL8 (with dependencies from EPEL).

Signed-off-by: Neal Gompa <ngompa13@gmail.com>

### CODE REVIEW

**Comment by Conan-Kudo** [comment]
@rjfd Note that I don't think EPEL quite provides everything yet for EL8 for this package, but this means you'll be ready for when it does.

File: sesdev.spec [hunk 1]
```diff
@@ -63,24 +63,16 @@ sed -i -e 's/^\s*lv.qemu_use_session = false$//g' seslib/templates/Vagrantfile.j
...
-%else
 %py3_install
-%endif
+%fdupes %{buildroot}%{python3_sitelib}
 
 %files
 %license LICENSE
 %doc CHANGELOG.md README.md
-%{python3_sitelib}/*
+%{python3_sitelib}/ses*
```
Comment (smithfarm) [inline sesdev.spec]: why is this needed?

File: sesdev.spec [hunk 2]
```diff
@@ -16,35 +16,35 @@
 #
 
 
+# Enable Python dependency generation on EL8
+%{?python_enable_dependency_generator}
```
Comment (smithfarm) [inline sesdev.spec]: do we need to put this inside an `%if` block?

```
%if 0%{?rhel} >= 8
...
%endif
```

File: sesdev.spec [hunk 3]
```diff
@@ -16,35 +16,35 @@
...
+Requires:       python3-click >= 6.7
 Requires:       python3-Jinja2 >= 2.10.1
 Requires:       python3-pycryptodomex >= 3.4.6
 Requires:       python3-PyYAML >= 3.13
 Requires:       python3-setuptools
-%else
-Requires:       python3-jinja2 >= 2.10.1
-Requires:       python3-pycryptodomex >= 3.4.6
-Requires:       python3-pyyaml >= 3.13
 %endif
```
Comment (smithfarm) [inline sesdev.spec]: I gather that for CentOS 8 the python dependencies will be generated automagically, but what about for CentOS 7?

File: sesdev.spec (hunk 1 above)
Comment (Conan-Kudo) [reply sesdev.spec]: Fedora Python Packaging Guidelines<sup>[1]</sup> state to not do a universal glob like that. Doing it this way also catches potentially accidents where `%python3_sitelib` is not expanded for some reason and bombs out instead of producing an incorrect package.

[1]: https://docs.fedoraproject.org/en-US/packaging-guidelines/Python/#_files_to_include

File: sesdev.spec (hunk 3 above)
Comment (Conan-Kudo) [reply sesdev.spec]: There's more work to be done here if you wanted to support CentOS 7. EPEL 7 doesn't have your dependencies packaged, so someone would need to step up and help get those dependencies packaged there first.

File: sesdev.spec (hunk 2 above)
Comment (Conan-Kudo) [reply sesdev.spec]: Not necessary. The `%{?` part of the macro declaration means that if it doesn't exist, it'll just do nothing.

Comment (smithfarm) [reply sesdev.spec]: Technically not necessary, perhaps, but I prefer to make things explicit. With the `%if` block I suggested, the comment could be dropped (I also tend to avoid comments in spec files...).

File: sesdev.spec (hunk 1 above)
Comment (smithfarm) [reply sesdev.spec]: I like that it makes the packaging somewhat more explicit, and it also elegantly avoids packaging the `__pycache__` directory that is caught up by the current line.

Could we make it even more explicit by writing:

```
%{python3_sitelib}/sesdev*
%{python3_sitelib}/seslib*
```

?

File: sesdev.spec (hunk 2 above)
Comment (Conan-Kudo) [reply sesdev.spec]: I just realized you're also building for Fedora 29, and it would be useful for that too. Unless you plan to drop F29 support?

File: sesdev.spec (hunk 1 above)
Comment (Conan-Kudo) [reply sesdev.spec]: @smithfarm Have you considered rearchitecting it so that there's no *.py files in the top level `%python3_sitelib`?

Comment (Conan-Kudo) [reply sesdev.spec]: @smithfarm I just pushed a change to move sesdev into a package instead of being a module. That should clean things up for you.

File: sesdev.spec (hunk 2 above)
Comment (rjfd) [reply sesdev.spec]: > Unless you plan to drop F29 support?

Negative. We should keep supporting F29

File: sesdev.spec (hunk 3 above)
Comment (rjfd) [reply sesdev.spec]: I don't think we will ever support centos 7

File: sesdev.spec (hunk 2 above)
Comment (smithfarm) [reply sesdev.spec]: `%if 0%{?rhel} >= 8 || 0%{?fedora} >= 29` in that case

File: sesdev.spec (hunk 1 above)
Comment (rjfd) [reply sesdev.spec]: > Have you considered rearchitecting it so that there's no *.py files in the top level %python3_sitelib?

What are the problems of having py files in the top level?

File: sesdev.spec (hunk 2 above)
Comment (Conan-Kudo) [reply sesdev.spec]: It's not needed for Fedora 30 or newer: https://fedoraproject.org/wiki/Changes/EnablingPythonGeneratorsByDefault

File: sesdev.spec (hunk 1 above)
Comment (Conan-Kudo) [reply sesdev.spec]: It means that you wind up with `__pycache__` at the top level too, which is annoying...

File: sesdev.spec (hunk 2 above)
Comment (rjfd) [reply sesdev.spec]: I'm in favor of being more explicit, even if not needed.

Comment (Conan-Kudo) [reply sesdev.spec]: Done.

**Comment by rjfd** [comment]
@Conan-Kudo are you building `sesdev` in any other packaging platform like copr or centos koji?

**Comment by Conan-Kudo** [comment]
Planning on building in COPR, why?

**Comment by Conan-Kudo** [comment]
Do you want to set up this project to auto release into COPR?

**Comment by rjfd** [comment]
> Planning on building in COPR, why?

No, I was just wondering if you were.

**Comment by Conan-Kudo** [comment]
@smithfarm You're supposed to package the `__pycache__` directories. Packages that don't are in violation of both Fedora and openSUSE packaging policy.
